packets.go

```diff
@@ -259,11 +259,9 @@ func (mc *mysqlConn) writeAuthPacket(cipher []byte) error {
 	}
 
 	// Calculate packet length and get buffer with that size
-	data := mc.buf.takeSmallBuffer(pktLen + 4)
-	if data == nil {
-		// can not take the buffer. Something must be wrong with the connection
-		errLog.Print(ErrBusyBuffer)
-		return driver.ErrBadConn
+	data, err := mc.buf.takeSmallBuffer(pktLen + 4)
+	if err != nil {
+		return err
```

Comment:
For example here it should be impossible that we're in a transaction with unread rows. Here the `driver.ErrBadConn` should still be returned as there is this is a brand new connection and there really shouldn't be any unread data in the buffer.